docs/reference/remap/sha2.cue

```diff
@@ -0,0 +1,56 @@
+package metadata
+
+remap: functions: sha2: {
+	arguments: [
+		{
+			name:        "value"
+			description: "The text to calculate the hash for."
+			required:    true
+			type: ["string"]
+		},
+		{
+			name:        "variant"
+			description: "The variant of the algorithm to use."
+			required:    false
+			default:     "SHA-512/256"
+			type: ["string"]
+		},
+	]
+	return: ["string"]
+	category: "hash"
+	description: #"""
+			Calculates a sha2 hash of a given string. The allowed variants are:
+			- SHA-224
+			- SHA-256
+			- SHA-384
+			- SHA-512
+			- SHA-512/224
+			- SHA-512/256
```

Comment:
A few of our functions take a string, but they limit what the string can be set to.

Can we somehow represent that explicitly in cue using the free-form description field?